File: 1128-RemoveAllAdjacentDuplicatesInString/1128-RemoveAllAdjacentDuplicatesInString.java
// Last updated: 8/5/2025, 2:55:44 PM
class Solution {
    public String removeDuplicates(String S) {
        Stack <Character> stack = new Stack<Character>();
        for(int i=0; i<S.length(); i++){
            if(!stack.isEmpty() && S.charAt(i) == stack.peek()){
                stack.pop();
            } else {
                stack.push(S.charAt(i));
            }
        }
        StringBuffer result = new StringBuffer();
        for(char c:stack){
            result.append(c);
        }
        return result.toString();
    }
}

File: 1128-RemoveAllAdjacentDuplicatesInString/1128-RemoveAllAdjacentDuplicatesInString.py
# Last updated: 8/5/2025, 2:55:42 PM
class Solution:
    def removeDuplicates(self, S: str) -> str:
        stack, index = [], 0

        while index < len(S):
            if len(stack) > 0 and S[index] == stack[-1]:
                stack.pop()
            else:
                stack.append(S[index])
            index += 1
        return ''.join(stack)